src/layers/scenegraph-layer/scenegraph-layer.js

```diff
@@ -0,0 +1,312 @@
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {ScenegraphLayer as DeckScenegraphLayer} from '@deck.gl/mesh-layers';
+import {load} from '@loaders.gl/core';
+import {GLTFScenegraphLoader} from '@luma.gl/addons';
+
+import Layer from '../base-layer';
+import memoize from 'lodash.memoize';
+import ScenegraphLayerIcon from './scenegraph-layer-icon';
+
+export const scenegraphRequiredColumns = ['lat', 'lng'];
+export const scenegraphOptionalColumns = ['altitude'];
+
+export const scenegraphPosAccessor = ({lat, lng, altitude}) => d => [
+  // lng
+  d.data[lng.fieldIdx],
+  // lat
+  d.data[lat.fieldIdx],
+  // altitude
+  altitude && altitude.fieldIdx > -1 ? d.data[altitude.fieldIdx] : 0
+];
+
+export const scenegraphPosResolver = ({lat, lng, altitude}) =>
+  `${lat.fieldIdx}-${lng.fieldIdx}-${altitude ? altitude.fieldIdx : 'z'}`;
+
+export const scenegraphVisConfigs = {
+  radius: 'radius',
+  fixedRadius: 'fixedRadius',
+  opacity: 'opacity',
+  colorRange: 'colorRange',
+  radiusRange: 'radiusRange',
+  //
+  sizeScale: 'sizeScale',
+  angleX: 'angleX',
+  angleY: 'angleY',
+  angleZ: 'angleZ'
+};
+
+export default class ScenegraphLayer extends Layer {
+  constructor(props) {
+    super(props);
+
+    this.registerVisConfig(scenegraphVisConfigs);
+    this.getPosition = memoize(scenegraphPosAccessor, scenegraphPosResolver);
+
+    // prepare layer info modal
+    // this._layerInfoModal = IconInfoModalFactory();
+    // this.getSvgIcons();
+  }
+
+  get type() {
+    return '3D';
+  }
+
+  get requiredLayerColumns() {
+    return scenegraphRequiredColumns;
+  }
+
+  get optionalColumns() {
+    return scenegraphOptionalColumns;
+  }
+
+  get columnPairs() {
+    return this.defaultPointColumnPairs;
+  }
+
+  get layerIcon() {
+    return ScenegraphLayerIcon;
+  }
+
+  get visualChannels() {
+    return {
+      ...super.visualChannels,
+      size: {
+        ...super.visualChannels.size,
+        range: 'radiusRange',
+        property: 'radius',
+        channelScaleType: 'radius'
+      }
+    };
+  }
+
+  get layerInfoModal() {
+    return {
+      id: 'iconInfo',
+      template: this._layerInfoModal,
+      modalProps: {
+        title: 'How to draw icons'
+      }
+    };
+  }
+
+  static findDefaultLayerProps({fieldPairs, fields}) {
```

Comment:
This function will get called when dataset is added to kepler,  it automatically create layers based on certain column name conversion. If you don't plan to automatically create a 3d modal layer, you don't need this static method

Reply:
Removing for now